homeassistant/helpers/template.py

```diff
@@ -478,6 +479,28 @@ def __repr__(self) -> str:
         return 'Template("' + self.template + '")'
 
 
+class RateLimit:
+    """Class to control update rate limits."""
+
+    def __init__(self, hass: HomeAssistantType):
+        """Initialize rate limit."""
+        self._hass = hass
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Optional[timedelta]:
+        """Handle a call to the class."""
+        delta = timedelta(*args, **kwargs)
+
+        render_info = self._hass.data.get(_RENDER_INFO)
+        if render_info is not None:
+            render_info.rate_limit = delta
+
+        return delta
```

Comment:
It seems a little weird that we wrap a timedelta. Most of the times that this is called, one might not want to use the return value, and now needs to use it in an expression (`and`) or assign it to an unused variable.

If we add a `__str__` that returns an empty value, we could support something like this:

```jinja2
{{ rate_limit(5) -}}

{{ value_json.measurement }}
```

but then maybe we should instead tell people to use `{% set _ = rate_limit(5) %}` and not even consider printing it.

Reply:
I don't think we ever want to print it, only use it to calculate something in the template.  That seems unlikely though. Will make adjustments

Reply:
I made it return an empty string. If we change the implementation under the hood it might end up with a breaking change later so best to return nothing.

Reply:
```
    tmp = template.Template(
        "{{ rate_limit(minutes=1) }}random", hass
    )

    info = tmp.async_render_to_info()
    assert info.result() == "random"
    assert info.rate_limit == timedelta(minutes=1)
```